### PR TITLE
BUG: Provide a better error message for out-of-order fields

### PR DESCRIPTION
**Pull request by eric-wieser**
Fixes gh-10389.

Marking as 1.14.1 since this was exposed by the structured field change

### CODE REVIEW

**Comment by charris** [comment]
Eric, are you basing these for backport? I ask because of the circleci failures.

**Comment by ahaldane** [comment]
LGTM

File: numpy/core/tests/test_multiarray.py [hunk 1]
```diff
@@ -6453,6 +6453,19 @@ def test_relaxed_strides(self):
             shape, strides = get_buffer_info(arr, ['C_CONTIGUOUS'])
             assert_(strides[-1] == 8)
 
+    def test_out_of_order_fields(self):
+        dt = np.dtype(dict(
+            formats=['<i4', '<i4'],
+            names=['one', 'two'],
+            offsets=[4, 0],
+            itemsize=8
+        ))
```
Comment (charris) [inline numpy/core/tests/test_multiarray.py]: These would look better on the previous line. Python indentation isn't like C blocks.

**Comment by charris** [comment]
Thanks Eric.

**Comment by eric-wieser** [comment]
> Eric, are you basing these for backport?

Yes

File: numpy/core/tests/test_multiarray.py (hunk 1 above)
Comment (eric-wieser) [reply numpy/core/tests/test_multiarray.py]: Used to writing things like this from javascript - for deeply nested parens, it makes things a lot clearer. Here you're right, it adds little value.

**Comment by charris** [comment]
I just do
```
git rebase HEAD^ --onto <new branch>
git push -f origin HEAD
```
and make the new PR. Adjust as fit for more than one commit.

**Comment by eric-wieser** [comment]
My goal here to to make a backport possible without touching the command line - I'll submit one now.

**Pull request merged**